plugins/flytekit-pydantic/tests/test_type_transformer.py

```diff
@@ -0,0 +1,167 @@
+
+
+class ConfigRequired(BaseModel):
+    """Config BaseModel for testing purposes with required attribute."""
+
+    model_config: Union[Dict[str, TrainConfig], TrainConfig]
+
+
+class ConfigWithFlyteFiles(BaseModel):
+    """Config BaseModel for testing purposes with flytekit.files.FlyteFile type hint."""
+
+    flytefiles: List[file.FlyteFile]
+
+
+class ConfigWithFlyteDirs(BaseModel):
+    """Config BaseModel for testing purposes with flytekit.files.FlyteFile type hint."""
+
+    flytedirs: List[directory.FlyteDirectory]
+
+
+class ChildConfig(Config):
+    """Child class config BaseModel for testing purposes."""
+
+    d: List[int] = [1, 2, 3]
+
+
+@pytest.mark.parametrize(
+    "python_type,kwargs",
+    [(Config, {}), (ConfigRequired, {"model_config": TrainConfig()}), (TrainConfig, {}), (TrainConfig, {})],
+)
+def test_transform_round_trip(python_type: Type, kwargs: Dict[str, Any]):
+    """Test that a (de-)serialization roundtrip results in the identical BaseModel."""
+    from flytekit.core.context_manager import FlyteContextManager
+
+    ctx = FlyteContextManager().current_context()
+
+    type_transformer = BaseModelTransformer()
+
+    python_value = python_type(**kwargs)
+
+    literal_value = type_transformer.to_literal(
+        ctx,
+        python_value,
+        python_type,
+        type_transformer.get_literal_type(python_value),
+    )
+
+    reconstructed_value = type_transformer.to_python_value(ctx, literal_value, type(python_value))
+
+    assert reconstructed_value == python_value
+    assert reconstructed_value.schema() == python_value.schema()
+
+
+@pytest.mark.parametrize(
+    "config_type,kwargs",
+    [
+        (Config, {"model_config": {"foo": TrainConfig(loss="mse")}}),
+        (ConfigRequired, {"model_config": {"foo": TrainConfig(loss="mse")}}),
+        (ConfigWithFlyteFiles, {"flytefiles": ["s3://foo/bar"]}),
+        (ConfigWithFlyteDirs, {"flytedirs": ["s3://foo/bar"]}),
+    ],
+)
+def test_pass_to_workflow(config_type: Type, kwargs: Dict[str, Any]):
+    """Test passing a BaseModel instance to a workflow works."""
+    cfg = config_type(**kwargs)
+
+    @task
+    def train(cfg: config_type) -> config_type:
+        return cfg
+
+    @workflow
+    def wf(cfg: config_type) -> config_type:
+        return train(cfg=cfg)
+
+    returned_cfg = wf(cfg=cfg)
+
+    assert cfg == returned_cfg
+
+
+@pytest.mark.parametrize(
+    "kwargs",
+    [
+        {"flytefiles": ["tests/folder/test_file1.txt", "tests/folder/test_file2.txt"]},
+    ],
+)
+def test_flytefiles_in_wf(kwargs: Dict[str, Any]):
+    """Test passing a BaseModel instance to a workflow works."""
+    cfg = ConfigWithFlyteFiles(**kwargs)
+
+    @task
+    def read(cfg: ConfigWithFlyteFiles) -> str:
+        with open(cfg.flytefiles[0], "r") as f:
+            return f.read()
+
+    @workflow
+    def wf(cfg: ConfigWithFlyteFiles) -> str:
+        return read(cfg=cfg)
+
+    string = wf(cfg=cfg)
+    assert string == "love sosa"
```

Comment:
My personal preference would be to not include any pop culture references but stick with e.g. "foo bar" which developers from any background will recognize as placeholders. Your call though.